book.rb

```diff
@@ -26,6 +27,10 @@ def to_json(*args)
   private
 
   def can_be_archived?
-    super || @cover_state == 'bad'
+    @archived = if super || @cover_state == 'bad'
+                  true
```

Comment:
optiona suggestion .. you can write 
super || @cover_state == 'bad'

Reply:
@bamsi  Thank you very much I will change it. 👍✅😊